qt/python/mantidqt/widgets/colorbar/colorbar.py

```diff
@@ -234,11 +242,15 @@ def update_clim(self):
         """
         if self.autoscale.isChecked():
             data = self.colorbar.mappable.get_array()
+            norm = NORM_OPTS[self.norm.currentIndex()]
             try:
                 try:
-                    self.cmin_value = data[~data.mask].min()
-                    self.cmax_value = data[~data.mask].max()
+                    masked_data = data[~data.mask]
```

Comment:
Not following what we're doing in this try block, particularly with the tilde operator (I understand it's the 2's complement of the thing it's operating on?)

Reply:
Good question, I haven't looked entirely at this, but if `data.mask` gives you the invalid indices, then `~data.mask` should give you the valid ones, so I think it’s just indexing into the array.

Reply:
Since I have some other nit picks, this method is starting to grow a little large - could we split it across the if block:

```
if self.autoscale.isChecked():
    self._autoscale_clim(...)
else:
    self._scale_clim_manual(...)
```

Then future readers can go to the appropriate method